WalletWasabi/WabiSabi/Client/CoinJoinManager.cs

```diff
@@ -281,14 +270,29 @@ bool TryRemoveTrackedAutoStart(IWallet wallet)
 
 		await WaitAndHandleResultOfTasksAsync(nameof(trackedAutoStarts), trackedAutoStarts.Values.Select(x => x.Task).ToArray()).ConfigureAwait(false);
 	}
-
+	
+	private bool TryRemoveTrackedAutoStart(ConcurrentDictionary<IWallet, TrackedAutoStart> trackedAutoStarts, IWallet wallet)
+	{
+		if (trackedAutoStarts.TryRemove(wallet, out var trackedAutoStart))
+		{
+			trackedAutoStart.CancellationTokenSource.Cancel();
+			trackedAutoStart.CancellationTokenSource.Dispose();
+			return true;
+		}
+		return false;
+	}
+	
 	private void ScheduleRestartAutomatically(IWallet walletToStart, ConcurrentDictionary<IWallet, TrackedAutoStart> trackedAutoStarts, bool stopWhenAllMixed, bool overridePlebStop, CancellationToken stoppingToken)
 	{
+		var delayRestart = 30;
 		if (trackedAutoStarts.ContainsKey(walletToStart))
 		{
-			walletToStart.LogDebug($"AutoStart was already scheduled.");
-			return;
+			walletToStart.LogDebug("AutoStart was already scheduled, cancel the last task and do not wait.");
+			TryRemoveTrackedAutoStart(trackedAutoStarts, walletToStart);
```

Comment:
You can just let this go in this case, it will be removed later? 

https://github.com/zkSNACKs/WalletWasabi/pull/10059/files#diff-28444f49c77fd0c2cd05f91515914d63fc3c9264ff1f24a56b2fc8e8167df850R317